src/extractor.rs

```diff
@@ -72,7 +73,7 @@ impl Extraction {
     }
 
     pub fn populate(
-        refs: Option<Vec<(String, Option<String>)>>,
+        refs: Option<Vec<(String, Option<String>, Option<RefVersion>)>>,
```

Comment:
GitHub linting rightfully suggests this type is particularly complex. Creating a new struct the ref type would help with readability here:

```suggestion
        refs: Option<Vec<Ref>>,
```

Reply:
I'm just now realizing that `Ref` is a commonly used type in Rust so you might want to consider a different name. Maybe `DbtRef` or something else to differentiate it.